pkg/reconciler/common/prune.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	openshiftPrefix  = "openshift"
+	kubePrefix       = "kube"
+	tektonSA         = "tekton-pipelines-controller"
+	CronName         = "resource-pruner"
+	JobsTKNImageName = "IMAGE_JOB_PRUNER_TKN"
+	defaultTKNImage  = "gcr.io/tekton-releases/dogfooding/tkn@sha256:f69a02ef099d8915e9e4ea1b74e43b7a9309fc97cf23cb457ebf191e73491677"
+)
+
+func Prune(k kubernetes.Interface, ctx context.Context, pru v1alpha1.Prune, targetNamespace string) error {
```

Comment:
let us accept ` v1alpha1.TektonComponent` as argument.
pass TektonConfig instance from when this function is called.

then take `targetNamespace` and `pruneParameters` from the spec.